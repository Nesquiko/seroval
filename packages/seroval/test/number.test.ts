import { describe, it, expect } from 'vitest';
import {
  serialize,
  serializeAsync,
  toJSON,
  toJSONAsync,
} from '../src';

describe('number', () => {
  describe('serialize', () => {
    it('supports numbers', () => {
      const value = 0xDEADBEEF;
      expect(serialize(value)).toBe(`${value}`);
      expect(serialize(NaN)).toBe('NaN');
      expect(serialize(Infinity)).toBe('1/0');
      expect(serialize(-Infinity)).toBe('-1/0');
      expect(serialize(-0)).toBe('-0');
    });
  });
  describe('serializeAsync', () => {
    it('supports numbers', async () => {
      const value = 0xDEADBEEF;
      expect(await serializeAsync(Promise.resolve(value))).toBe(`Promise.resolve(${value})`);
      expect(await serializeAsync(Promise.resolve(NaN))).toBe('Promise.resolve(NaN)');
      expect(await serializeAsync(Promise.resolve(Infinity))).toBe('Promise.resolve(1/0)');
      expect(await serializeAsync(Promise.resolve(-Infinity))).toBe('Promise.resolve(-1/0)');
      expect(await serializeAsync(Promise.resolve(-0))).toBe('Promise.resolve(-0)');
    });
  });
  describe('toJSON', () => {
    it('supports numbers', () => {
      const value = 0xDEADBEEF;
      expect(JSON.stringify(toJSON(value))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(NaN))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Infinity))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(-Infinity))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(-0))).toMatchSnapshot();
    });
  });
  describe('toJSONAsync', () => {
    it('supports numbers', async () => {
      const value = 0xDEADBEEF;
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(value))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(NaN))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Infinity))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(-Infinity))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(-0))),
      ).toMatchSnapshot();
    });
  });
});
