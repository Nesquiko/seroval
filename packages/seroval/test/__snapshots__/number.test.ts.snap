// Vitest Snapshot v1

exports[`number > toJSON > supports numbers 1`] = `"{\\"t\\":{\\"t\\":0,\\"s\\":3735928559},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`number > toJSON > supports numbers 2`] = `"{\\"t\\":{\\"t\\":8},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`number > toJSON > supports numbers 3`] = `"{\\"t\\":{\\"t\\":6},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`number > toJSON > supports numbers 4`] = `"{\\"t\\":{\\"t\\":7},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`number > toJSON > supports numbers 5`] = `"{\\"t\\":{\\"t\\":5},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`number > toJSONAsync > supports numbers 1`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":0,\\"s\\":3735928559}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`number > toJSONAsync > supports numbers 2`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":8}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`number > toJSONAsync > supports numbers 3`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":6}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`number > toJSONAsync > supports numbers 4`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":7}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`number > toJSONAsync > supports numbers 5`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":5}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;
